controller/src/main/java/io/pravega/controller/store/stream/PersistentStreamBase.java

```diff
@@ -1683,6 +1691,9 @@ boolean compareWriterPositions(Map<Long, Long> position1, Map<Long, Long> positi
                         if (txnRecord.getCommitOrder() == order) {
                             // if entry matches record's position then include it
                             transactionsMap.put(txnId, txnRecord);
+                            if (transactionsMap.size() >= limit) {
```

Comment:
do we require this check given that the txnIds are not greater than the passed limit ?

Reply:
transactionsMap is actually accumulating filtered txns from the txnIds across all iterations and the limit s really to be imposed on the number of entries we put in this map as that will be sent back to the caller to use as the batch of transactions